Zend/Optimizer/dce.c

```diff
@@ -538,18 +539,18 @@ int dce_optimize_op_array(zend_op_array *op_array, zend_ssa *ssa, bool reorder_d
 
 	/* We have no dedicated phi vector, so we use the whole ssa var vector instead */
 	ctx.instr_worklist_len = zend_bitset_len(op_array->last);
-	ctx.instr_worklist = alloca(sizeof(zend_ulong) * ctx.instr_worklist_len);
+	ctx.instr_worklist = do_alloca(sizeof(zend_ulong) * ctx.instr_worklist_len, use_heap);
```

Comment:
I don't think you can reuse the same use_heap variable for all allocas, because they might make different choices.

I think it would be better to replace this with use of arena allocation and a checkpoint, that's what we usually do in optimizations.